### PR TITLE
Use flutter-cirrus gke cluster for framework tasks.

### PR DESCRIPTION
**Pull request by godofredoc**
Start using flutter-cirrus gke cluster for linux framework tasks. This is in preparation to enable alerts, metrics and monitoring for the different flutter repositories.

Bug: https://github.com/flutter/flutter/issues/77624

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [X] I signed the [CLA].
- [X] I listed at least one issue that this PR fixes in the description above.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt.
- [X] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by christopherfujino** [comment]
Looking at the cirrus docker pre-build failure, I think it's a flake (I'm able to curl the file locally). Triggered a re-run https://cirrus-ci.com/task/5513951426379776

**Comment by christopherfujino** [comment]
Hmm, so the re-run got further, but I suspect this failure will need either a dockerfile change or a config change.

```
debconf: unable to initialize frontend: Dialog
debconf: (TERM is not set, so the dialog frontend is not usable.)
debconf: falling back to frontend: Readline
debconf: unable to initialize frontend: Readline
debconf: (This frontend requires a controlling tty.)
debconf: falling back to frontend: Teletype
```

This issue has several possible fixes, setting `ENV TERM linux` in the dockerfile seems promising.

**Comment by godofredoc** [comment]
> Hmm, so the re-run got further, but I suspect this failure will need either a dockerfile change or a config change.
> 
> ```
> debconf: unable to initialize frontend: Dialog
> debconf: (TERM is not set, so the dialog frontend is not usable.)
> debconf: falling back to frontend: Readline
> debconf: unable to initialize frontend: Readline
> debconf: (This frontend requires a controlling tty.)
> debconf: falling back to frontend: Teletype
> ```
> 
> [This issue](https://github.com/phusion/baseimage-docker/issues/58) has several possible fixes, setting `ENV TERM linux` in the dockerfile seems promising.

Thanks I made the updates and now everything is working as expected.

**Comment by godofredoc** [comment]
This PR is now ready for review.

File: .cirrus.yml [hunk 1]
```diff
@@ -24,11 +21,19 @@ environment:
...
+gcp_credentials: ENCRYPTED[1917385f09764ac58515f6fee14ed58688afc808849b244969d41d0b082061eecd0a4c8ba801f00833877a12b51f8bb5]
+
+
 # LINUX SHARDS
 task:
-  container:
-    # https://cirrus-ci.org/guide/docker-builder-vm/#dockerfile-as-a-ci-environment
+  gke_container:
     dockerfile: "dev/ci/docker_linux/Dockerfile"
+    builder_image_name: docker-builder # gce vm image
```
Comment (christopherfujino) [inline .cirrus.yml:32]: I'm guessing this is the same VM that we use to build docker images for the engine?

Comment (godofredoc) [reply .cirrus.yml:32]: Yes it is the same. The idea is to make all the repos use Cirrus in a similar way and encapsulate them to a single service account and gcp project.

Comment (christopherfujino) [reply .cirrus.yml:32]: SGTM

**Review by christopherfujino — APPROVED** [review]
LGTM

**Comment by fkorotkov** [comment]
Hey @godofredoc,

FYI there are some issues on your GKE cluster. Pod status for [this task](https://cirrus-ci.com/task/4941454465302528) constantly reports that it's waiting with this message:

> failed to reserve container name "main_cirrus-ci-task-4941454465302528_default_ca22180a-a7a2-4432-98d6-d8a32732a9a9_0": name "main_cirrus-ci-task-4941454465302528_default_ca22180a-a7a2-4432-98d6-d8a32732a9a9_0" is reserved for "da923bfa5ed0775ec8a3a0fd1fb32702271b3a18bac2d51c02daad54f0a37adb"

Seems this issue is related to https://github.com/containerd/containerd/issues/4604. Just wanted to let you know.

**Pull request merged**